### PR TITLE
#284 Use chained calls from mockk for ImagesRepositoryTest and PartiesRepositoryTest

### PR DESCRIPTION
**Pull request by kmaslowiec**

### CODE REVIEW

File: app/src/test/kotlin/com/intive/picover/images/repository/ImagesRepositoryTest.kt [hunk 1]
```diff
@@ -36,13 +35,12 @@ class ImagesRepositoryTest : ShouldSpec(
...
 				val uriResult: Uri = mockk()
+
 				every { imageReference.listAll() } returns mockk {
-					coEvery { await() } returns listResult
+					coEvery { await().items } returns listOf(storageItem)
 				}
+
 				every { storageItem.downloadUrl } returns mockk {
 					coEvery { await() } returns uriResult
 				}
```
Comment (michal-kucznerowicz) [inline app/src/test/kotlin/com/intive/picover/images/repository/ImagesRepositoryTest.kt]: These every blocks can be merged. Same in "throw exception WHEN fetching the uri fails" test

**Review by michal-kucznerowicz — CHANGES_REQUESTED** [review]

File: app/src/test/kotlin/com/intive/picover/images/repository/ImagesRepositoryTest.kt (hunk 1 above)
Comment (kmaslowiec) [reply app/src/test/kotlin/com/intive/picover/images/repository/ImagesRepositoryTest.kt]: Good point. Updated.

**Review by michal-kucznerowicz — APPROVED** [review]

File: app/src/test/kotlin/com/intive/picover/images/repository/ImagesRepositoryTest.kt [hunk 2]
```diff
@@ -36,15 +34,16 @@ class ImagesRepositoryTest : ShouldSpec(
 
 		should("downloaded URIs WHEN storage reference and uri is fetched successfully") {
 			runTest(testDispatcher) {
-				val listResult: ListResult = mockk {
-					every { items } returns listOf(storageItem)
-				}
 				val uriResult: Uri = mockk()
+
```
Comment (MarekMacko) [inline app/src/test/kotlin/com/intive/picover/images/repository/ImagesRepositoryTest.kt:38]: Why it's separated? 
`uriResult` declaration and mocked behavior of `imageReference` in line 39 belongs to "given" block. According to give-when-then approach.

File: app/src/test/kotlin/com/intive/picover/images/repository/ImagesRepositoryTest.kt [hunk 3]
```diff
@@ -36,15 +34,16 @@ class ImagesRepositoryTest : ShouldSpec(
...
 				every { imageReference.listAll() } returns mockk {
-					coEvery { await() } returns listResult
-				}
-				every { storageItem.downloadUrl } returns mockk {
-					coEvery { await() } returns uriResult
+					coEvery { await().items } returns listOf(
```
Comment (MarekMacko) [inline app/src/test/kotlin/com/intive/picover/images/repository/ImagesRepositoryTest.kt:40]: You can also merge `imageReference.listAll()` with `await().items` there is no need to create a redundant mockk object, so the final form would be 
```
coEvery { imageReference.listAll().await().items }
```
Please apply above suggestion to other tests

Comment (kmaslowiec) [reply app/src/test/kotlin/com/intive/picover/images/repository/ImagesRepositoryTest.kt:40]: Please see: https://github.com/Android-Guild/Picover/pull/322

Comment (MarekMacko) [reply app/src/test/kotlin/com/intive/picover/images/repository/ImagesRepositoryTest.kt:40]: Please check my last sentence, "Please apply above suggestion to other tests"

Comment (kmaslowiec) [reply app/src/test/kotlin/com/intive/picover/images/repository/ImagesRepositoryTest.kt:40]: Done. Please continue the conversation in https://github.com/Android-Guild/Picover/pull/315 or Tag me. Thank you.

**Pull request merged**